### PR TITLE
Feat : Added Reducer and context to pomodora app

### PR DESCRIPTION
**Pull request by Omkar-Chaskar**
Added context provider and reducers for
- Add Note
- Edit note
- delete note
- archive / unarchive note
- pin / unpin note
- restore note

### CODE REVIEW

File: src/style/addnote.css [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  justify-content: center;
+  align-items: center;
+  gap: 1rem;
+}
+
+.tags-input {
+  display: flex;
+  align-items: flex-start;
+  flex-wrap: wrap;
+  min-height: 48px;
```
Comment (Megha-Pathak) [inline src/style/addnote.css:42]: Using rems would be better here. 

Throughout code try to consistent with rems or px. Right now it's a mix of both. 

Rest of the code looks well structured. Good work.

**Review by Megha-Pathak — APPROVED** [review]

File: src/App.js [hunk 2]
```diff
@@ -1,43 +1,31 @@
-import "./App.css";
-import logo from "./logo.png";
+import "./styles.css";
+import { Routes, Route } from "react-router-dom";
+import Header from "./component/header";
```
Comment (VedanthB) [inline src/App.js:3]: You can do a barrel import/export here , make the code look cleaner

File: src/component/edit.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
+import "../styles.css";
+import { Link, useParams } from "react-router-dom";
+import { useNotes } from "../context/noteProvider";
+// import { useState } from "react";
```
Comment (VedanthB) [inline src/component/edit.js:4]: remove comments

**Review by VedanthB — APPROVED** [review]
Good work, 

The code is clean and easy to read. Variable naming is good.

File: src/App.js [hunk 4]
```diff
@@ -1,43 +1,31 @@
-import "./App.css";
-import logo from "./logo.png";
+import "./styles.css";
```
Comment (HeyNitin) [inline src/App.js:1]: Try to use Absolute imports from here on

File: src/component/archive.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+            state.archive.map((note) => {
+              return (
+                <li className="card" key={note.id}>
+                  <div className="card__primary-action card__primary-action-column">
+                    {state.pin &&
+                    state.pin.some((item) => note.id === item.id) ? (
+                      <MdPushPin
+                        name="pin item"
+                        title="Pin"
+                        className="badge-up-right-corner p3 text-grey ptr"
```
Comment (HeyNitin) [inline src/component/archive.js:38]: There is inconsistency in class names, above it's joined with '__' and here with '-', if there is no specific reason, try to be consistent in CSS styling

File: src/component/edit.js (hunk 3 above)
Comment (HeyNitin) [inline src/component/edit.js:4]: You should remove commented code

File: src/component/pomodora.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        </div>
+      </div>
+      <div className="note-detail-container card">
+        <div className="note-container">
+          <div className="note-header">
+            <h4>Note Title...</h4>
+          </div>
+          <div className="note-body">
+            <p className="bold p2 text-grey">
+              Lorem Ipsum es simplemente el texto de relleno de las imprentas y
```
Comment (HeyNitin) [inline src/component/pomodora.js:28]: content can be exported to another file and then be imported from there

File: src/style/addnote.css [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+.tags-input:focus-within {
+  border: 1px solid #0052cc;
+}
+
+input,
+textarea {
+  flex: 1;
+  border: none;
+  height: 46px;
+  font-size: 14px;
```
Comment (HeyNitin) [inline src/style/addnote.css:57]: try to be consistent in css, some places you're using rem and px at others

**Review by HeyNitin — APPROVED** [review]
CSS should be consistent and so should be class names,
Left some comments, mainly about coding practices
Rest is good

**Pull request merged**